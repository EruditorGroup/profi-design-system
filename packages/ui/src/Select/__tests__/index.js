// yarn test src/client/ui/select/custom/__tests__/index.js
import * as React from 'react';
import CustomSelect from 'client/ui/select/custom';
import {shallow} from 'enzyme';
import {mockedComponent} from 'utilities/jest/mocks';

jest.unmock('client/ui/select/custom');
jest.unmock('client/ui/select/custom/components/option');
jest.unmock('client/ui/utils/useVisibilityToggler');

jest.mock('client/ui/renderToBodyDiv', () =>
  mockedComponent('RenderToBodyDiv'),
);
jest.mock('client/ui/relativeToElementPosition', () =>
  mockedComponent('RelativeToElementPosition'),
);

describe('CustomSelect', () => {
  let data;

  beforeEach(() => {
    data = {
      shmid: 'TEST-shmid',
      className: 'TEST-className',
      disabled: false,
      value: 'TEST-value',
      placeholder: 'TEST-placeholder',
      prefix: 'test-prefix',
      onChange: () => {},
      options: [
        {
          label: 'TEST-label-1',
          value: 'TEST-value-1',
          disabled: false,
        },
        {
          label: 'TEST-label-2',
          value: 'TEST-value-2',
          disabled: true,
        },
        {
          label: 'TEST-label-3',
          value: 'TEST-value-3',
          disabled: true,
        },
      ],
    };
  });

  it('render CustomSelect', () => {
    React.useState = jest.fn(() => [true, jest.fn()]);
    expect(
      shallow(
        <CustomSelect {...data} setIsOpen={jest.fn()} setValue={jest.fn()} />,
      ),
    ).toMatchSnapshot();
  });

  it('Открыли select list', () => {
    React.useState = jest.fn(() => [true, jest.fn()]);
    const component = shallow(
      <CustomSelect {...data} setIsOpen={jest.fn()} setValue={jest.fn()} />,
    );
    component.find('.ui-custom-select__input').simulate('click');
    expect(component).toMatchSnapshot();
  });

  it('Открыли select list и выбрали первое значение', () => {
    const onChange = jest.fn();
    const component = shallow(
      <CustomSelect {...data} setIsOpen={jest.fn()} onChange={onChange} />,
    );
    component.find('.ui-custom-select__input').simulate('click');
    component
      .find('.ui-custom-select__option')
      .first()
      .prop('onClick')('TEST-value-1');
    expect(onChange).toBeCalledWith('TEST-value-1');
  });

  it('Открыли select list и закрыли', () => {
    React.useState = jest.fn(() => [true, jest.fn()]);
    const component = shallow(
      <CustomSelect {...data} setIsOpen={jest.fn()} setValue={jest.fn()} />,
    );
    component.find('.ui-custom-select__input').simulate('click');
    component.find('.ui-custom-select__input').simulate('click');
    expect(component).toMatchSnapshot();
  });
});
