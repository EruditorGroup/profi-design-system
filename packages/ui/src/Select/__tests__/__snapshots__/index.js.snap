// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect render CustomSelect 1`] = `
<Fragment>
  <div
    className="TEST-className select-module__root-MMDOP select-module__display-inline-block-3o3J8"
    data-shmid="TEST-shmid"
  >
    <div
      className="select-module__input-IhK3M ui-custom-select__input"
      onClick={[Function: onClick]}
    >
      <div
        className="select-module__prefix-2AFEh"
      >
        test-prefix
      </div>
      <div>
        TEST-value
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`CustomSelect Открыли select list 1`] = `
<Fragment>
  <div
    className="TEST-className select-module__root-MMDOP select-module__opened-1_OE6 select-module__display-inline-block-3o3J8"
    data-shmid="TEST-shmid"
  >
    <div
      className="select-module__input-IhK3M ui-custom-select__input"
      onClick={[Function: onClick]}
    >
      <div
        className="select-module__prefix-2AFEh"
      >
        test-prefix
      </div>
      <div>
        TEST-value
      </div>
    </div>
  </div>
  <Mocked:RenderToBodyDiv>
    <Mocked:RelativeToElementPosition
      parentElement={
        Object {
          "current": null,
        }
      }
    >
      <ForwardRef(Scrollable)
        className="select-module__options-266z9 select-module__options_custom-11twF"
      >
        <Memo(SelectOption)
          className="ui-custom-select__option"
          design="custom"
          disabled={false}
          key="TEST-value-1_0"
          label="TEST-label-1"
          onClick={[Function: handleOptionClick]}
          value="TEST-value-1"
        />
        <Memo(SelectOption)
          className="ui-custom-select__option"
          design="custom"
          disabled={true}
          key="TEST-value-2_1"
          label="TEST-label-2"
          onClick={[Function: handleOptionClick]}
          value="TEST-value-2"
        />
        <Memo(SelectOption)
          className="ui-custom-select__option"
          design="custom"
          disabled={true}
          key="TEST-value-3_2"
          label="TEST-label-3"
          onClick={[Function: handleOptionClick]}
          value="TEST-value-3"
        />
      </ForwardRef(Scrollable)>
    </Mocked:RelativeToElementPosition>
  </Mocked:RenderToBodyDiv>
</Fragment>
`;

exports[`CustomSelect Открыли select list и закрыли 1`] = `
<Fragment>
  <div
    className="TEST-className select-module__root-MMDOP select-module__display-inline-block-3o3J8"
    data-shmid="TEST-shmid"
  >
    <div
      className="select-module__input-IhK3M ui-custom-select__input"
      onClick={[Function: onClick]}
    >
      <div
        className="select-module__prefix-2AFEh"
      >
        test-prefix
      </div>
      <div>
        TEST-value
      </div>
    </div>
  </div>
</Fragment>
`;
